### PR TITLE
Fix rawtypes warning for Iterator in platform cluster

### PR DESCRIPTION
**Pull request by mklaehn**
There are several places where the compiler warns about rawtypes used for Iterator

   [repeat] /Users/martin/Repositories/git/netbeans/platform/openide.util/src/org/openide/util/TopologicalSortException.java:358: warning: [rawtypes] found raw type: Iterator
   [repeat]         public Iterator edges() {
   [repeat]                ^
   [repeat]   missing type arguments for generic class Iterator<E>
   [repeat]   where E is a type-variable:
   [repeat]     E extends Object declared in interface Iterator

Convert this code to declare available types or similar ways to reduce/remove these warnings

### CODE REVIEW

File: platform/core.multiview/src/org/netbeans/core/multiview/MultiViewPeer.java [hunk 1]
```diff
@@ -801,11 +799,11 @@ private UndoRedo privateGetUndoRedo() {
      * Delegates to CloseOperationHandler.
      */
     boolean canClose() {
-        Collection col = model.getCreatedElements();
-        Iterator it = col.iterator();
-        Collection badOnes = new ArrayList();
+        Collection<MultiViewElement> col = model.getCreatedElements();
+        Iterator<MultiViewElement> it = col.iterator();
+        Collection<CloseOperationState> badOnes = new ArrayList<>();
         while (it.hasNext()) {
```
Comment (svenreimers) [inline platform/core.multiview/src/org/netbeans/core/multiview/MultiViewPeer.java]: Should this be a nested for loop not to while loops?

File: platform/openide.util/src/org/openide/util/NbCollections.java [hunk 2]
```diff
@@ -63,11 +63,9 @@ private NbCollections() {}
      *         to the named type (or they may be null)
      * @throws ClassCastException if some entry in the raw set was not well-typed, and only if <code>strict</code> was true
      */
-    public static <E> Set<E> checkedSetByCopy(Set rawSet, Class<E> type, boolean strict) throws ClassCastException {
-        Set<E> s = new HashSet<E>(rawSet.size() * 4 / 3 + 1);
-        Iterator it = rawSet.iterator();
-        while (it.hasNext()) {
-            Object e = it.next();
+    public static <E> Set<E> checkedSetByCopy(Set<?> rawSet, Class<E> type, boolean strict) throws ClassCastException {
```
Comment (matthiasblaesing) [inline platform/openide.util/src/org/openide/util/NbCollections.java]: According to the documentation in this case `rawSet` is intentionally not a generic set. Wouldn't it be better to suppress warning in this case?

File: platform/openide.util/src/org/openide/util/NbCollections.java [hunk 3]
```diff
@@ -91,11 +89,9 @@ private NbCollections() {}
      *         to the named type (or they may be null)
      * @throws ClassCastException if some entry in the raw list was not well-typed, and only if <code>strict</code> was true
      */
-    public static <E> List<E> checkedListByCopy(List rawList, Class<E> type, boolean strict) throws ClassCastException {
-        List<E> l = (rawList instanceof RandomAccess) ? new ArrayList<E>(rawList.size()) : new LinkedList<E>();
-        Iterator it = rawList.iterator();
-        while (it.hasNext()) {
-            Object e = it.next();
+    public static <E> List<E> checkedListByCopy(List<?> rawList, Class<E> type, boolean strict) throws ClassCastException {
```
Comment (matthiasblaesing) [inline platform/openide.util/src/org/openide/util/NbCollections.java]: Same comment as line 66 applies.

File: platform/openide.util/src/org/openide/util/NbCollections.java [hunk 4]
```diff
@@ -120,11 +116,9 @@ private NbCollections() {}
      *         to the named types (or they may be null)
      * @throws ClassCastException if some key or value in the raw map was not well-typed, and only if <code>strict</code> was true
      */
-    public static <K,V> Map<K,V> checkedMapByCopy(Map rawMap, Class<K> keyType, Class<V> valueType, boolean strict) throws ClassCastException {
-        Map<K,V> m2 = new HashMap<K,V>(rawMap.size() * 4 / 3 + 1);
-        Iterator it = rawMap.entrySet().iterator();
-        while (it.hasNext()) {
-            Map.Entry e = (Map.Entry) it.next();
+    public static <K,V> Map<K,V> checkedMapByCopy(Map<?, ?> rawMap, Class<K> keyType, Class<V> valueType, boolean strict) throws ClassCastException {
```
Comment (matthiasblaesing) [inline platform/openide.util/src/org/openide/util/NbCollections.java]: Comment from line 66 applies.

File: platform/openide.util/src/org/openide/util/NbCollections.java [hunk 5]
```diff
@@ -191,7 +188,7 @@ public void remove() {
      *               if true, {@link ClassCastException} may be thrown from an iterator operation
      * @return an iterator guaranteed to contain only objects of the requested type (or null)
      */
-    public static <E> Iterator<E> checkedIteratorByFilter(Iterator rawIterator, final Class<E> type, final boolean strict) {
+    public static <E> Iterator<E> checkedIteratorByFilter(Iterator<?> rawIterator, final Class<E> type, final boolean strict) {
```
Comment (matthiasblaesing) [inline platform/openide.util/src/org/openide/util/NbCollections.java]: Same comment as line 66.

File: platform/openide.util/src/org/openide/util/NbCollections.java [hunk 6]
```diff
@@ -222,8 +219,8 @@ protected boolean accept(Object o) {
      *               if true, a {@link ClassCastException} may arise during some set operation
      * @return a view over the raw set guaranteed to match the specified type
      */
-    public static <E> Set<E> checkedSetByFilter(Set rawSet, Class<E> type, boolean strict) {
-        return new CheckedSet<E>(rawSet, type, strict);
+    public static <E> Set<E> checkedSetByFilter(Set<?> rawSet, Class<E> type, boolean strict) {
```
Comment (matthiasblaesing) [inline platform/openide.util/src/org/openide/util/NbCollections.java]: Same comment as line 66.

File: platform/openide.util/src/org/openide/util/TopologicalSortException.java [hunk 7]
```diff
@@ -47,7 +47,7 @@
     /** vertexes sorted by increasing value of y */
     private Stack<Vertex> dualGraph = new Stack<Vertex>();
 
-    TopologicalSortException(Collection vertexes, Map<?,? extends Collection<?>> edges) {
+    TopologicalSortException(Collection<?> vertexes, Map<?,? extends Collection<?>> edges) {
```
Comment (matthiasblaesing) [inline platform/openide.util/src/org/openide/util/TopologicalSortException.java]: Is the switch from raw type to generic intended? If so, is the bound planned as is?

File: platform/openide.util/src/org/openide/util/MapFormat.java [hunk 8]
```diff
@@ -88,7 +88,7 @@
     * For common work use  <code>format(pattern, arguments) </code>.
     * @param arguments keys and values to use in the format
     */
-    public MapFormat(Map arguments) {
+    public MapFormat(Map<String, ? extends Object> arguments) {
```
Comment (matthiasblaesing) [inline platform/openide.util/src/org/openide/util/MapFormat.java]: Just to be sure: this changes/fixates the generic bounds of an exported type - intentionally?

File: platform/core.multiview/src/org/netbeans/core/multiview/MultiViewPeer.java (hunk 1 above)
Comment (mklaehn) [reply platform/core.multiview/src/org/netbeans/core/multiview/MultiViewPeer.java]: It really shouldn't. They could be converted to enhanced-for-loops but they shouldn't be nested.
The first while loop collects all non-closeable MVEs which the second while loop - if the closeHandler indicated such - attemps to close again.

You know the dialog showing you the unsaved files when you attempt to close one editor or the whole IDE. That's the principle here.

**Comment by matthiasblaesing** [comment]
There are merge conflicts, unanswered comments and this is nearly two years old. Closing.

**Pull request closed without merging**